data/tools/unit_tree/wiki_output.py

```diff
@@ -16,29 +16,31 @@ def main():
 
     punits = {}
 
-    defines = "NORMAL,ENABLE_ARMAGEDDON_DRAKE,ENABLE_DWARVISH_ARCANISTER," +\
+    base_defines = "NORMAL,ENABLE_ARMAGEDDON_DRAKE,ENABLE_DWARVISH_ARCANISTER," +\
         "ENABLE_DWARVISH_RUNESMITH,ENABLE_ANCIENT_LICH,ENABLE_DEATH_KNIGHT," +\
         "ENABLE_TROLL_SHAMAN,ENABLE_WOLF_ADVANCEMENT,ENABLE_WOSE_SHAMAN," +\
-        "ENABLE_PARAGON,ENABLE_SAURIAN_SPEARTHROWER,DISABLE_GRAND_MARSHAL"
+        "ENABLE_PARAGON,ENABLE_SAURIAN_SPEARTHROWER,DISABLE_GRAND_MARSHAL".split(",")
 
     sys.stderr.write("Parsing core units...\n")
-    wesnoth.parser.parse_text("{core/units.cfg}", defines)
+    wesnoth.parser.parse_text("{core/units.cfg}", ",".join(base_defines))
     punits["mainline"] = wesnoth.parser.get_all(tag = "units")
     punits["mainline"] += wesnoth.parser.get_all(tag = "+units")
 
     all_campaigns = {}
     sys.stderr.write("Parsing campaigns...\n")
-    wesnoth.parser.parse_text("{campaigns}", defines)
+    wesnoth.parser.parse_text("{campaigns}", ",".join(base_defines))
     campaigns = wesnoth.parser.get_all(tag = "campaign")
     for campaign in campaigns:
+        campaign_defines = [] + base_defines
         define = campaign.get_text_val("define")
         ed = campaign.get_text_val("extra_defines")
-        if ed: define += "," + ed
+        if define is not None: campaign_defines += define.split(",")
```

Comment:
I could be mistaken, but I don't believe `define=` accepts a comma-separated list?

Reply:
Sorry. Noted.